tests/playbooks/tests_infiniband.yml

```diff
@@ -0,0 +1,160 @@
+          changed_when: false
+          when: network_provider == 'nm'
+
+        - name: Assert that the settings in the virtual device ib0.000a match
+            the specified settings
+          assert:
+            that:
+              - virtual_device_setting.stdout is search("ipoib pkey 0x800a mode
+                datagram")
+            msg: "the settings in the virtual device ib0.000a match the
+              specified settings"
+          when: network_provider == 'nm'
+
+        - command: ip -4 a s ib0.000a
+          name: "** TEST check IPv4"
+          register: result
+          until: "'198.51.100.133/30' in result.stdout"
+          retries: 20
+          delay: 2
+          when: network_provider == 'nm'
+
+        - command: ip -6 a s ib0.000a
+          name: "** TEST check IPv6"
+          register: result
+          until: "'2001:db8::2/32' in result.stdout"
+          retries: 20
+          delay: 2
+          when: network_provider == 'nm'
+
+        - name: Get the details for the virtual device ib0.800a
+          command: ip -d a s ib0.800a
+          register: virtual_device_setting
+          ignore_errors: yes
+          changed_when: false
+          when: network_provider == 'initscripts'
+
+        - name: Assert that the settings in the virtual device ib0.800a match
+            the specified settings
+          assert:
+            that:
+              - virtual_device_setting.stdout is search("ipoib pkey 0x800a mode
+                datagram")
+            msg: "the settings in the virtual device ib0.800a match the
+              specified settings"
+          when: network_provider == 'initscripts'
+
+        - command: ip -4 a s ib0.800a
+          name: "** TEST check IPv4"
+          register: result
+          until: "'198.51.100.133/30' in result.stdout"
+          retries: 20
+          delay: 2
+          when: network_provider == 'initscripts'
+
+        - command: ip -6 a s ib0.800a
+          name: "** TEST check IPv6"
+          register: result
+          until: "'2001:db8::2/32' in result.stdout"
+          retries: 20
+          delay: 2
+          when: network_provider == 'initscripts'
+      always:
+        - block:
+            - name: Import network role
+              import_role:
+                name: linux-system-roles.network
+              vars:
+                network_connections:
+                  - name: ib0
+                    persistent_state: absent
+                    state: down
+                  - name: ib0-10
+                    persistent_state: absent
+                    state: down
+              ignore_errors: true
+            - command: ip link del ib0.000a
+              ignore_errors: true
+              when: network_provider == 'nm'
+            - command: ip link del ib0.800a
+              ignore_errors: true
+              when: network_provider == 'initscripts'
+          tags:
+            - "tests::cleanup"
+  tags:
+    - "tests::expfail"
```

Comment:
On second thought, I think the tag should be explicit - so use `tests::infiniband` instead so that we know the purpose of the tag

Reply:
Updated. Thanks.